hummingbot/connector/exchange/coinzoom/coinzoom_exchange.py

```diff
@@ -589,8 +600,35 @@ async def _update_balances(self):
         """
         Calls REST API to update total and available balances.
         """
-        account_info = await self._api_request("GET", Constants.ENDPOINT["USER_BALANCES"], is_auth_required=True)
-        self._process_balance_message(account_info)
+        try:
+            # Check for in progress balance updates, queue if fetching and none already waiting, otherwise skip.
+            if self._update_balances_fetching:
+                if not self._update_balances_queued:
+                    self._update_balances_queued = True
+                    await self._update_balances_finished.wait()
+                    self._update_balances_queued = False
+                    self._update_balances_finished = asyncio.Event()
+                else:
+                    return
+            self._update_balances_fetching = True
+            account_info = await self._api_request("GET", Constants.ENDPOINT["USER_BALANCES"], is_auth_required=True)
+            self._process_balance_message(account_info)
+            self._update_balances_fetching = False
+            # Set balance update finished event if there's one waiting.
+            if self._update_balances_queued and not self._update_balances_finished.is_set():
+                self._update_balances_finished.set()
+        except Exception as e:
+            if self._update_balances_queued:
+                if self._update_balances_finished.is_set():
+                    self._update_balances_finished = asyncio.Event()
+                else:
+                    self._update_balances_finished.set()
+                self._update_balances_queued = False
+            if self._update_balances_fetching:
+                self._update_balances_fetching = False
+            warn_msg = (f"Could not fetch balance update from {Constants.EXCHANGE_NAME}")
+            self.logger().network(f"Unexpected error while fetching balance update - {str(e)}", exc_info=True,
+                                  app_warning_msg=warn_msg)
```

Comment:
```suggestion
"""
I'll propose a suggestion to simplify this. Since balance updates will be applied with inflight order snapshot, I suggest we make this function to only execute once per time. i.e no need to queue an extra call.
"""
```

Reply:
I simplified this because I believe we don't need an extra queue since balance will eventually be calculated with snapshot.